### ISSUE TITLE
(node:1954) DeprecationWarning: GridStore is deprecated, and will be removed in a future version. Please use GridFSBucket instead

### ISSUE THREAD

**Issue by psonhoang** [devconcept/multer-gridfs-storage#11]
Hi there,

I just ran into this warning after updating to v.3.1.0.

(node:1954) DeprecationWarning: GridStore is deprecated, and will be removed in a future version. Please use GridFSBucket instead

**Comment by devconcept** [comment]
That's strange. This module only use GridStore if GridFSBucket is not available. The new version didn't change anything regarding streaming routines.

Do you use the gridfs-stream module in your app?

**Comment by psonhoang** [comment]
Yes i do use gridfs-stream, it seems to pop up the error when it reads from the database

**Comment by devconcept** [comment]
Hi there. The `gridfs-stream` module exists because `GridStore` is really bad handling streams. The `GridFSBucket` class was created to solve that problem. Just drop the gridfs-stream dependency and replace it with the native MongoDb class.

**Comment by devconcept** [comment]
Is the [GridFsBucket](http://mongodb.github.io/node-mongodb-native/2.1/api/GridFSBucket.html) class. You can access it by using

```javascript
const {GridFSBucket} = require('mongo')
```

Check the docs to know how to use it.

**Comment by qwtfy** [comment]
really simple article i found
https://node.dev/post/node-is-simple-part-4
